### PR TITLE
Switch environment to use Pipenv

### PR DESCRIPTION
**Pull request by rcoh**
1. Switch from requirements.txt to `pipenv`. This DRY's the dependencies -- the runtime deps are in setup.py. The dev deps are in the Pipfile.
2. Bake the latest git tag into the version string and use it from the CLI. This removes the need to ever manually bump a version for Python. To support Nuitka, a `semgrep/version.py` file is created as the wheel is built. Homebrew versioning support comes from the gitarchive.

I verified that this produces the correct metadata for homebrew by pointing homebrew `head` at this branch and doing a `homebrew install --HEAD`

### CODE REVIEW

File: semgrep/Pipfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+pytest = "*"
+wheel = "*"
+tox = "*"
+pytest-snapshot = "*"
+
+[packages]
+semgrep = {editable = true,path = "."}
+
+[requires]
+python_version = "3.7"
```
Comment (underyx) [inline semgrep/Pipfile:16]: We run tests on 3.6, couldn't we set that here?

**Comment by underyx** [comment]
Could you update the makefile as well?

File: semgrep/Pipfile (hunk 1 above)
Comment (rcoh) [reply semgrep/Pipfile:16]: a bunch of things (nuitka, docker images, etc.) currently rely on 3.7 so keeping it for now

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
       - name: Install and test semgrep-core
         run: eval $(opam  env --root /home/opam/.opam --set-root) && cd semgrep-core && opam install -y . && make all && make test && make install
       - name: Install and test semgrep
-        run: eval $(opam  env --root /home/opam/.opam --set-root) && cd semgrep && export PATH=/github/home/.local/bin:$PATH && pip3 install wheel==0.34.2 && pip3 install -r requirements.txt && make test && make qa-test && make all
+        run: eval $(opam  env --root /home/opam/.opam --set-root) && cd semgrep && export PATH=/github/home/.local/bin:$PATH && pip3 install wheel==0.34.2 pipenv && pipenv install --dev && make test && make qa-test && make all
```
Comment (brendongo) [inline .github/workflows/build.yml]: Is there a reason to have wheel installed separately instead of adding it to the --dev pipenv?

Comment (rcoh) [reply .github/workflows/build.yml]: I think no, let me try moving it.

**Review by brendongo — APPROVED** [review]

**Pull request merged**